src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -235,7 +235,15 @@ internal virtual TypeEnumerator GetTypeEnumerator(Type type, string assemblyFile
                 {
                     if (discoveryOption == TestDataSourceDiscoveryOption.DuringDiscovery)
                     {
-                        if (DynamicDataAttached(sourceLevelParameters, test, tests))
+                        Lazy<TestMethodInfo?> testMethodInfo = GetTestMethodInfo(sourceLevelParameters, test);
+
+                        // Add fixture tests like AssemblyInitialize, AssemblyCleanup, ClassInitialize, ClassCleanup.
+                        if (MSTestSettings.CurrentSettings.FixturesEnabled && testMethodInfo.Value is not null)
```

Comment:
Also this is called within the foreach of each test case we have found which is not good.